### PR TITLE
RockPaperScissors Pettingzoo

### PR DESCRIPTION
**Pull request by rodrigodelazcano**
## Why are these changes needed?

Miscellaneous maintenance to Rock-Paper-Scissors scissors example code, replaces environment itself with upstream PettingZoo environment (it has identical functionality with more minor fixes). Upstreaming the environment removes maintenance on your end.
## Related issue number

This PR is an updated version of PR [#13993](https://github.com/ray-project/ray/pull/13993), taking into consideration the newer rllib and pettingzoo rps environment versions.
## Checks

-  [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by justinkterry** [comment]
@richard4912 This is a redo of a PR that went without review after 5 months. Could this please be looked into this like?

File: rllib/examples/rock_paper_scissors_multiagent.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 """A simple multi-agent env with two agents playing rock paper scissors.
-
```
Comment (richardliaw) [inline rllib/examples/rock_paper_scissors_multiagent.py:2]: We should leave this here for standard pydoc

File: rllib/examples/rock_paper_scissors_multiagent.py [hunk 2]
```diff
@@ -109,27 +121,28 @@ def select_policy(agent_id, episode, **kwargs):
...
     if args.as_test:
         raise ValueError(
             "Desired reward difference ({}) not reached! Only got to {}.".
-            format(args.stop_reward, env.player1_score - env.player2_score))
+            format(args.stop_reward, reward_diff))
 
 
 def run_with_custom_entropy_loss(args, stop):
     """Example of customizing the loss function of an existing policy.
-
```
Comment (richardliaw) [inline rllib/examples/rock_paper_scissors_multiagent.py:132]: same here - leave this line space for pydocstring format.

**Review by richardliaw — APPROVED** [review]
@rodrigodelazcano thanks a bunch!

**Comment by richardliaw** [comment]
BTW, @jkterry1 sorry, this PR must have slipped through. Also you tagged "richard4912" who is not part of the Ray project; be sure to tag "richardliaw" in the future.

Thanks!

File: rllib/examples/rock_paper_scissors_multiagent.py [hunk 3]
```diff
@@ -68,35 +81,34 @@ def run_same_policy(args, stop):
 
 def run_heuristic_vs_learned(args, use_lstm=False, trainer="PG"):
     """Run heuristic policies vs a learned agent.
```
Comment (richardliaw) [inline rllib/examples/rock_paper_scissors_multiagent.py:83]:
```suggestion
    """Run heuristic policies vs a learned agent.

```

File: rllib/examples/rock_paper_scissors_multiagent.py [hunk 4]
```diff
@@ -109,27 +121,28 @@ def select_policy(agent_id, episode, **kwargs):
...
     # Reward (difference) not reached: Error if `as_test`.
     if args.as_test:
         raise ValueError(
             "Desired reward difference ({}) not reached! Only got to {}.".
-            format(args.stop_reward, env.player1_score - env.player2_score))
+            format(args.stop_reward, reward_diff))
 
 
 def run_with_custom_entropy_loss(args, stop):
     """Example of customizing the loss function of an existing policy.
```
Comment (richardliaw) [inline rllib/examples/rock_paper_scissors_multiagent.py:146]:
```suggestion
    """Example of customizing the loss function of an existing policy.

```

File: rllib/examples/rock_paper_scissors_multiagent.py [hunk 5]
```diff
@@ -1,5 +1,4 @@
 """A simple multi-agent env with two agents playing rock paper scissors.
```
Comment (richardliaw) [inline rllib/examples/rock_paper_scissors_multiagent.py:1]:
```suggestion
"""A simple multi-agent env with two agents playing rock paper scissors.

```

**Comment by rodrigodelazcano** [comment]
Hi @richardliaw! Thank you for the review. Will this changes do to make the final merge?

**Comment by richardliaw** [comment]
ah it looks like lint is still failing - have you run the linting script?

**Comment by rodrigodelazcano** [comment]
@richardliaw is the lint still failing?

**Pull request merged**